packages/idyll-document/src/utils/index.js

```diff
@@ -95,17 +95,25 @@ const getVars = (arr, context) => {
     const [, [, nameValue]] = nameArr
     const [, [valueType, valueValue]] = valueArr;
 
-    if (valueType === 'value') acc[nameValue] = valueValue;
-    if (valueType === 'variable') acc[nameValue] = context[valueValue];
-    if (valueType === 'expression') {
-      const expr = valueValue;
-
-      acc[nameValue] = {
-        value: evalExpression(context, expr),
-        update: (newState, oldState) => {
-          return evalExpression(Object.assign({}, oldState, newState), expr)
+    switch(valueType) {
+      case 'value':
+        acc[nameValue] = valueValue;
+        break;
+      case 'variable':
+        if (context.hasOwnProperty(valueValue)) {
+          acc[nameValue] = context[valueValue];
+        } else {
+          acc[nameValue] = evalExpression(context, expr);
+        }
+        break;
+      case 'expression':
+        const expr = valueValue;
+        acc[nameValue] = {
+          value: evalExpression(context, expr),
+          update: (newState, oldState) => {
+            return evalExpression(Object.assign({}, oldState, newState), expr)
+          }
```

Comment:
This could be a one line change couldn't it? Either way is fine though.

`if (valueType === 'variable') acc[nameValue] = context[valueValue] || evalExpression(context, expr);`

Reply:
Yeah, I made that change, just because I thought there'd be more logic to add in. I can simplify it back (although I'm a little wary of that type of conditional assignment when 0 is a valid value for first expression)

Reply:
Ah, good point about 0. As you were... :)